pkg/steps/lease.go

```diff
@@ -64,8 +71,10 @@ func (s *leaseStep) Provides() api.ParameterMap {
 	if parameters == nil {
 		parameters = api.ParameterMap{}
 	}
-	parameters[leaseEnv] = func() (string, error) {
-		return s.leasedResource, nil
+	for i, l := range s.leases {
+		parameters[l.Env] = func() (string, error) {
+			return s.leases[i].resource, nil
+		}
```

Comment:
```diff
diff --git a/pkg/steps/lease.go b/pkg/steps/lease.go
index b3363bc6..2869cb8d 100644
--- a/pkg/steps/lease.go
+++ b/pkg/steps/lease.go
@@ -71,9 +71,9 @@ func (s *leaseStep) Provides() api.ParameterMap {
 	if parameters == nil {
 		parameters = api.ParameterMap{}
 	}
-	for _, l := range s.leases {
+	for i, l := range s.leases {
 		parameters[l.Env] = func() (string, error) {
-			return l.resource, nil
+			return s.leases[i].resource, nil
 		}
 	}
 	return parameters
```

Go <3

Reply:
That's not going to work either, is it?

Reply:
```diff
diff --git a/pkg/steps/lease.go b/pkg/steps/lease.go
index b3363bc6..f3fa795a 100644
--- a/pkg/steps/lease.go
+++ b/pkg/steps/lease.go
@@ -71,7 +71,8 @@ func (s *leaseStep) Provides() api.ParameterMap {
 	if parameters == nil {
 		parameters = api.ParameterMap{}
 	}
-	for _, l := range s.leases {
+	for i := range s.leases {
+		l := &s.leases[i]
 		parameters[l.Env] = func() (string, error) {
 			return l.resource, nil
 		}
```
Go <3 <3

Reply:
(a test that would catch https://github.com/openshift/ci-tools/pull/1259#discussion_r494168399 cannot be created because of https://github.com/openshift/ci-tools/pull/1232#discussion_r492717389, I'll add one when we have configurable leases)